src/Sites/twitch.tv/Runtime/VideoChatListener.ts

```diff
@@ -0,0 +1,204 @@
+import { Observable } from 'rxjs';
+import { filter, takeUntil, tap } from 'rxjs/operators';
+import { Logger } from 'src/Logger';
+import { BaseTwitchChatListener } from 'src/Sites/twitch.tv/Runtime/BaseChatListener';
+import { TwitchPageScript } from 'src/Sites/twitch.tv/twitch';
+import { MessagePatcher } from 'src/Sites/twitch.tv/Util/MessagePatcher';
+import { Twitch } from 'src/Sites/twitch.tv/Util/Twitch';
+
+
+/**
+ * Chat listener specifically for handling VODs, clips, and other videos.
+ * Unfortunately, Twitch's React components and DOM for video chats
+ * are too different from the ones used for live stream chats,
+ * therefore requiring this listener.
+ */
+export class TwitchVideoChatListener extends BaseTwitchChatListener {
+
+	constructor(page: TwitchPageScript) {
+		super(page);
+	}
+
+	start(): void {
+		const listener = this;
+
+		// Handle changes only if FFZ isn't enabled.
+		if (!this.page.ffzMode) {
+
+			Logger.Get().debug('Twitch video chat rerender detected, rendering 7TV emotes');
+			this.renderAll(listener.twitch.getVideoMessages());
+		}
+	}
+
+	listen(): void {
+
+		// Let FFZ handle things if it's available.
+		if (this.page.ffzMode) {
+			return;
+		}
+
+		Logger.Get().info('Listen for new video chat messages.');
+
+		this.observeDOM().pipe(
+			takeUntil(this.killed),
+			filter(message => !!message.component),  // Ignore messages with no component states, like subs/resubs.
+			tap(message => {
+				this.renderPaintOnNametag(message);
+				this.onMessage(message);
+			})
+		).subscribe();
+	}
+
+	private renderAll(messages: Twitch.VideoMessageAndComponent[]): void {
+		for (const message of messages) {
+			if (!message.component?.props) {
+				continue;
+			}
+
+			this.onMessage(message);
+			this.renderPaintOnNametag(message);
+		}
+	}
+
+	renderPaintOnNametag(message: Twitch.VideoMessageAndComponent) {
```

Comment:
Some of these methods could be part of the abstract class, perhaps with an argument to handle minor differences between live/video chats. For example move the bulk of the logic in this method to `BaseTwitchChatListener` then call `super.renderPaintOnNametag()` from the extending class.

Reply:
Yup. I can do that.